lib/mix/lib/mix/tasks/help.ex

```diff
@@ -94,13 +94,26 @@ defmodule Mix.Tasks.Help do
     Mix.raise("Unexpected arguments, expected \"mix help --search PATTERN\"")
   end
 
-  def run([task]) do
-    loadpaths!()
-    opts = Application.get_env(:mix, :colors)
-    opts = [width: width(), enabled: ansi_docs?(opts)] ++ opts
+  def run([task_or_module]) do
+    if Regex.match?(~r/(:|[A-Z]).*/, task_or_module) do
+      if Mix.Project.get() do
+        Mix.Tasks.Compile.run([])
+      end
 
-    for doc <- verbose_doc(task) do
-      print_doc(task, doc, opts)
+      task_or_module
+      |> Code.string_to_quoted!()
+      |> IEx.Introspection.decompose(__ENV__)
+      |> Code.eval_quoted()
+      |> elem(0)
+      |> IEx.Introspection.h()
+    else
+      loadpaths!()
+      opts = Application.get_env(:mix, :colors)
+      opts = [width: width(), enabled: ansi_docs?(opts)] ++ opts
```

Comment:
We need to apply this configuration to IEx as well, so maybe we need something like:

```elixir
iex_colors = Application.get_env(:iex, :colors, [])

try do
  Application.put_env(:iex, :colors, Application.get_env(:mix, :colors))
  IEx.Introspection.h()
after
  Application.put_env(:iex, :colors, iex_colors)
end
```